### ISSUE TITLE
Alternative logic: scan for a time VS on/off

### ISSUE THREAD

**Issue by Mrnofish** [rowrawer/stream-detector#78]
The way I use the extension, I usually don't want it to be harvesting links all the time: instead, I turn it on, leave it that way for a while, and then turn it off.

Of course I forget to turn it off all the time!

It would be nice to have an option for the extension to look for links for a while and then turn itself off automatically. Perhaps this could work well with #63

**Comment by rowrawer** [comment]
I'll take it into consideration, but it does seem like quite a bit of work for something that can be done in 2 clicks.

**Comment by Mrnofish** [comment]
It does sound like it should be easy to remember to turn it the extension off, in practice I regularly find it swamped by hundreds and hundreds of links, and it happens to me regularly.

I know a little Javascript but have no hands-on time with extensions, could you please give me some pointers so I can better appreciate the extent of the changes required ?

**Comment by rowrawer** [comment]
I think your best bet would simply be to look through the code here on GitHub, honestly. It's not that it's too hard to implement - would most likely be based on the setTimeout function - it's just that I don't know whether it'd really be worth implementing as the off button is right there in the menu. I'll be taking a look at implementing a blacklist feature as mentioned in #79, I think that could help alleviate the issue somewhat.

**Comment by Mrnofish** [comment]
> I'll be taking a look at implementing a blacklist feature as mentioned in #79, I think that could help alleviate the issue somewhat.

I see where you're coming from, but in practice I suspect the impact would be very small; If anything, a whitelist might work better, but even then that wouldn't avoid hundreds of links from piling up, and so ultimately everything would still revolve around the Disable option.

> I think your best bet would simply be to look through the code here on GitHub, honestly

I did take a look, however it seems a tad too advanced for the time being.

I'm guessing a big part of it, is beginning to understand how a browser extension works to begin with, but currently, between a personal project, and trying to study JavaScript, I've got my hands full, so it'll have to wait.
